tests/validation/tests/v3_api/test_node.py

```diff
@@ -31,6 +44,230 @@ def test_add_node_label():
+    node_id = nodes[0].id
+    node = client.by_id_node(node_id)
+    node_name = node.nodeName
+    command = "label nodes " + node_name + " " + test_label + "=" + label_value
+    print(command)
+    execute_kubectl_cmd(command, False)
+    time.sleep(2)
+
+    # Label should be added
+    wait_for_condition(client, node, check_label_added(test_label), None, 10)
+    node = client.reload(node)
+    node_labels = node.labels.data_dict()
+    assert node_labels[test_label] == "bar", "Label is NOT added on the node"
+
+    # check via kubectl if label is set on the node
+    command = " get nodes " + node_name
+    print(command)
+    node_detail = execute_kubectl_cmd(command)
+    print(node_detail["metadata"]["labels"])
+    assert test_label in node_detail["metadata"]["labels"], \
+        "Label is not set in kubectl"
+    assert node_detail["metadata"]["labels"][test_label] == label_value
+
+    # remove label
+    del node_labels[test_label]
+    client.update(node, labels=node_labels)
+    time.sleep(2)
+
+
+def test_node_label_add_ui_delete_kubectl():
+    test_label = "foo"
+    label_value = "bar"
+    client = cluster_detail["client"]
+    cluster = cluster_detail["cluster"]
+    create_kubeconfig(cluster)
+    nodes = client.list_node(clusterId=cluster.id).data
+    assert len(nodes) > 0
+    node_id = nodes[0].id
+    node = client.by_id_node(node_id)
+    node_name = node.nodeName
+
+    node_labels = node.labels.data_dict()
+    node_labels[test_label] = "bar"
+    client.update(node, labels=node_labels)
+    time.sleep(2)
+
+    # Label should be added
+    wait_for_condition(client, node, check_label_added(test_label), None, 10)
+    node = client.reload(node)
+    node_labels = node.labels.data_dict()
+    assert node_labels[test_label] == "bar", "Label is NOT added on the node"
+
+    # check via kubectl if label is set on the node
+    command = " get nodes " + node_name
+    print(command)
+    node_detail = execute_kubectl_cmd(command)
+    print(node_detail["metadata"]["labels"])
+    assert test_label in node_detail["metadata"]["labels"], \
+        "Label is not set in kubectl"
+    assert node_detail["metadata"]["labels"][test_label] == label_value
+
+    # delete through kubectl
+    command = " label node " + node_name + " " + test_label + "-"
+    execute_kubectl_cmd(command, False)
+    time.sleep(2)
+
+    # check via kubectl if label is NOT set on the node
+    command = " get nodes " + node_name
+    print(command)
+    node_detail = execute_kubectl_cmd(command)
+    print(node_detail["metadata"]["labels"])
+    assert test_label not in node_detail["metadata"]["labels"], \
+        "Label is set, seen via kubectl"
+
+    # Label should be deleted
+    wait_for_condition(client, node, check_label_removed(test_label), None, 10)
+    node = client.reload(node)
+    node_labels = node.labels.data_dict()
+    assert test_label not in node_labels, "Label is NOT deleted"
+
+
+def test_node_label_add_kubectl_delete_add():
```

Comment:
do you mean added by kubectl deleted by UI ?

Reply:
changed the test names.